hydromt_sfincs/workflows/merge.py

```diff
@@ -88,9 +88,20 @@ def merge_multi_dataarrays(
         method = "bilinear"
 
     if da_like is not None:  # reproject first raster to destination grid
-        da1 = da1.raster.reproject_like(da_like, method=method).load()
+        da1 = da1.raster.clip_bbox(
+            da_like.raster.box.to_crs(da1.raster.crs.to_epsg()).total_bounds, buffer=10
+        )
+        if np.any(np.array(da1.shape) <= 2):
+            # no data in da1 so use an empty array like da_like
+            logger.debug("No data da1, start with nan array")
+            da1 = xr.full_like(da_like, np.nan)
+        else:
+            da1 = da1.load()
```

Comment:
These load statement were necessary for the subgrid determination right? Because they are still sometimes causing memory issues

Reply:
Not sure. Would be good to take some time to profile these function to understand what works best, which load / clip statements are required. My understanding is that the way the data is stored is also very important to avoid memory errors. Perhaps we should plan an afternoon to work on this.